prometheus_hardware_exporter/collectors/perccli.py

```diff
@@ -0,0 +1,132 @@
+"""Collector for PowerEdgeRAID controller."""
+import socket
+from logging import getLogger
+from typing import Any, Dict, List
+
+from ..utils import Command, Result, get_json_output
+
+logger = getLogger(__name__)
+
+
+class PercCLI(Command):
+    """Command line tool for PowerEdge RAID Controller."""
+
+    prefix = "sudo"
+    command = "/opt/MegaRAID/perccli/perccli64"
+
+    def _get_controllers(self) -> Result:
+        result = self("/call show j")
+        if result.error:
+            logger.error(result.error)
+            return result
+
+        assert isinstance(result.data, str)
+        result = get_json_output(result.data)
+        if result.error:
+            return result
+        return result
+
+    def _cmd_status(self, controller: Dict[str, Any]) -> bool:
+        if controller["Command Status"]["Status"] == "Success":
+            return True
+        return False
+
+    def _ctrl_exists(self, controllers: Dict[str, Any]) -> bool:
+        if (
+            controllers["Controllers"][0]["Command Status"]["Status"] == "Failure"
+            and controllers["Controllers"][0]["Command Status"]["Description"]
+            == "No Controller found"
+        ):
+            return False
+        return True
+
+    def _get_ctrl_id(self, controller: Dict[str, Any]) -> int:
+        return controller["Command Status"]["Controller"]
+
+    def success(self) -> bool:
+        """Return false if command fail or controller not exists."""
+        return self.ctrl_exists()
+
+    def ctrl_exists(self) -> bool:
+        """Check if controller exists."""
+        result = self._get_controllers()
+        if result.error:
+            logger.error(result.error)
+            return False
+        assert isinstance(result.data, dict)
```

Comment:
Nonblocking: It might not be desired to use `assert` or raise error for any collector in the exporter, especially when there are other collectors because other collectors might be still functioning. Also, since the assertion is a type checking and we are already enforcing typing checking with `mypy`, why not consider using `mypy`? Same for other `assert`s

Reply:
Change the structure.